src/components/song/song.service.ts

```diff
@@ -315,4 +353,50 @@ export class SongService {
       total: result.total,
     };
   }
+
+  async listScriptureReferences(
+    songId: string,
+    session: ISession
+  ): Promise<ScriptureRange[]> {
+    const query = this.db
+      .query()
+      .match([
+        node('song', 'Song', {
+          id: songId,
+          active: true,
+          owningOrgId: session.owningOrgId,
+        }),
+        relation('out', '', 'scriptureReferences'),
+        node('scriptureRanges', 'ScriptureRange', { active: true }),
+      ])
+      .with('collect(scriptureRanges) as items')
+      .return('items');
```

Comment:
There's no need to collect() & first() row just to run() it. 
```suggestion
      .return('scriptureRanges')
      .asResult<{ scriptureRanges: Node<{ start: number; end: number }> }>();
```
Then
```tsx
const results = await query.run();
return results.map(item => 
  verseToScriptureRange({
    start: item.properties.start,
    end: item.properties.end,
  })
);